src/layer.rs

```diff
@@ -1,102 +1,732 @@
-use serde::Serialize;
+use std::fmt;
 
-#[derive(Debug, Clone, Default, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Layer {
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::{closure::Closure, JsValue};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Expression<T> {
+    Enum(T),
+    String(String),
+    Number(f64),
+    Bool(bool),
+    List(Vec<Expression<T>>),
+}
+
+pub trait EnumMarker {}
+impl<T: EnumMarker> From<T> for Expression<T> {
+    fn from(item: T) -> Self {
+        Expression::Enum(item)
+    }
+}
+impl<T> From<&str> for Expression<T> {
+    fn from(value: &str) -> Self {
+        Expression::String(value.into())
+    }
+}
+
+impl<T> From<String> for Expression<T> {
+    fn from(value: String) -> Self {
+        Expression::String(value)
+    }
+}
+
+impl<T> From<f64> for Expression<T> {
+    fn from(value: f64) -> Self {
+        Expression::Number(value)
+    }
+}
+
+impl<T> From<bool> for Expression<T> {
+    fn from(value: bool) -> Self {
+        Expression::Bool(value)
+    }
+}
+
+impl<T> From<Vec<&str>> for Expression<T> {
+    fn from(value: Vec<&str>) -> Self {
+        Expression::List(value.into_iter().map(|s| s.into()).collect())
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub enum Visibility {
+    #[default]
+    Visible,
+    None,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GetLayer {
```

Comment:
This is kinda sad but the return value of [map.getLayer()](https://docs.mapbox.com/mapbox-gl-js/api/map/#map#getlayer) is not really a normal layer such as you would get inside [map.getStyle()](https://docs.mapbox.com/mapbox-gl-js/api/map/#map#getstyle). Instead it is it's own wonky thing that is not really parsable.. So I made this getLayer struct in such a way that the two mapbox examples that use `getLayer` are implementable but that's it.. There is a [bigger comment about it here](https://github.com/yukinarit/mapbox-gl-rs/pull/64/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R902-R909) as well.